include/boost/json/basic_parser.hpp

```diff
@@ -197,10 +197,16 @@ reserve()
     // Reserve the largest stack we need,
     // to avoid reallocation during suspend.
     st_.reserve(
-        sizeof(state) + // document parsing state
-        (sizeof(state) * depth()) + // array and object state
-        sizeof(state) + // value parsing state
-        sizeof(state)); // comment/utf8 state
+        // document parsing state
+        sizeof(state) + 
+        // array and object state + size
+        (sizeof(state) + sizeof(std::size_t)) * depth() +
+        // value parsing state
+        sizeof(state) + 
+        // string size
+        sizeof(std::size_t) +
+        // comment/utf8 state
+        sizeof(state)); 
```

Comment:
this is harder to read please put the comments back on the same line. reformat line 203 thusly:
```
(sizeof(state) +
    sizeof(std::size_t)) * depth() +  // array and object state + size
```